### ISSUE TITLE
CreateProperties don't call in class inherited from DefaultContractResolver

### ISSUE THREAD

**Issue by Alex-1357** [JamesNK/Newtonsoft.Json#2620]
Hello, I publish my question with full source code in this topic https://stackoverflow.com/questions/70220551/protected-overloads-overrides-function-dont-working-in-my-vb-net-code-object-o?noredirect=1#comment124131780_70220551

Maybe this is not an issue of VB, this is something in your class, if you don't understand VB, this is the same code in C#, I have replaced VB code to C# code from other library, but result not changing, method CreateProperties don't call.

```csharp
namespace JSONNET
{
    public class JsonNetCustomAttributeProcessing<T> : Newtonsoft.Json.Serialization.DefaultContractResolver where T : Attribute
    {
        private Type _AttributeToIgnore = null;

        public JsonNetCustomAttributeProcessing()
        {
            _AttributeToIgnore = typeof(T);
        }

        protected override IList<JsonProperty> CreateProperties(Type Type1, MemberSerialization memberSerialization)
        {
            // Return MyBase.CreateProperties(Type1, memberSerialization)
            var List1 = Type1.GetProperties().Where(x => !x.GetCustomAttributes(true).Any(a => a.GetType() == _AttributeToIgnore)).Select(p => new JsonProperty()
            {
                PropertyName = p.Name,
                PropertyType = p.PropertyType,
                Readable = true,
                Writable = true,
                ValueProvider = base.CreateMemberValueProvider(p)
            }).ToList();
            return List1;
        }
    }
}
```

**Comment by elgonzo** [comment]
Who knows, maybe you made a mistake setting up the serializer.
Please provide a _minimal_ and _complete_ compilable code example demonstrating the issue.

**Comment by Alex-1357** [comment]
Now serializer configured as 
````
 Dim StringResult As String = JsonConvert.SerializeObject(JMNUserReport, New JsonSerializerSettings With {.ContractResolver = New JsonNetCustomAttributeProcessing(Of ParseServerResponse.AdminOnly), .Formatting = Formatting.Indented})
````
I have full published code, except data - class to be serialized, attribute definition, call serializer - there are nothing else. Is it not enough? Need add add test and create full compilable code? Ok, I try to load full project to dotnetfiddle.

**Comment by elgonzo** [comment]
> I have full published code, except data - class to be serialized, attribute definition, call serializer - there are nothing else. Is it not enough?

No, unfortunately, it is not enough. The implementation of your `JsonNetCustomAttributeProcessing<T>` as shown in your report above does not indicate any problem. Therefore, the problem must be caused by something else in your project _outside_ of the `JsonNetCustomAttributeProcessing<T>` class, and hence why i asked for a minimal and complete compilable code example demonstrating the issue.

If i use your VB.NET code from your StackOverflow question together with what you wrote in your last comment:
```
Dim StringResult As String = JsonConvert.SerializeObject(JMNUserReport, New JsonSerializerSettings With {.ContractResolver = New JsonNetCustomAttributeProcessing(Of ParseServerResponse.AdminOnly), .Formatting = Formatting.Indented})
```
the `CreateProperties` override in your `JsonNetCustomAttributeProcessing<T>` class **will** be called successfully. Dotnetfiddle proof: https://dotnetfiddle.net/wS6aRH (note that i augmented the _CreateProperties_ method with some console output, so you can see that it actually is being invoked.)

This leaves me with only one conclusion: Whatever the problem in your project, it is not with the `JsonNetCustomAttributeProcessing<T>` class itself, but something else...

**Comment by shueseman** [comment]
I am having the same issue as Alex reported.  Created custom resolver, with override of CreateProperties.  Create JSON Serializer, passing in settings object which uses the custom resolver.  The constructor on my custom resolver class IS getting called, but my breakpoint inside my overriden CreateProperties method never gets hit.   This is under .NET 6, Newtonsoft version 13.0.1.

I literally tried copy pasting "working" examples on the web with overriden CreateProperties and CreateProperty methods, and they don't work.

**Comment by elgonzo** [comment]
@shueseman

i am afraid that unless you provide a minimal and **complete** example (preferably on [https://dotnetfiddle.net](https://dotnetfiddle.net/wS6aRH) or similar online C# runtime environments) that reproduces, nobody will be able to provide meaningful assistance and suggestions of how to fix your issue.

Please also open a new issue report when you are providing your minimal and complete example (because there it is not certain at this point in time whether you really are having the same issue as reported here or whether you actually have a different issue that just so happens to superficially exhibits the same or similar observable behavior. (The rationale behind this is that it is relatively easy and little work to mark an issue report as a duplicate of another issue report, but it can be more work to separate two issues reported in the same issue report thread into two separate issue reports. Also, opening a separate issue report prevents "spamming" Alex-1357 -- the opener of this issue report -- with email notifications he probably might not be interested in.)

**Comment by shueseman** [comment]
Here is your complete code sample:

using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using Newtonsoft.Json.Serialization;
using System.Reflection;

namespace TestJsonContractResolver
{
    internal class Program
    {
        static void Main(string[] args)
        {
            var inputString = "{ 'id': '10', 'partitionKey': '10', 'coll1': [ 'jsonString4' ], 'coll2': [ 'jsonString5' ], 'coll3': [ 'jsonString6' ], '_rid': '0npSANR4lw8BAAAAAAAAAA == ', '_self': 'dbs / 0npSAA ==/ colls / 0npSANR4lw8 =/ docs / 0npSANR4lw8BAAAAAAAAAA ==/ ', '_etag': 'e600ec9e-0000-0100-0000-62f112480000', '_attachments': 'attachments / ', '_ts': 1659966024 }";

            var item = JObject.Parse(inputString);            

            var resolver = new DynamicContractResolver();

            var settings = new JsonSerializerSettings() { ContractResolver = resolver, Formatting = Formatting.Indented };

            var jsonString = Newtonsoft.Json.JsonConvert.SerializeObject(item, settings);
            
        }
    }

    public class DynamicContractResolver : DefaultContractResolver
    {
        private readonly char _startingWithChar = '_';
              

        protected override IList<JsonProperty> CreateProperties(Type type, MemberSerialization memberSerialization)
        {
            IList<JsonProperty> properties = base.CreateProperties(type, memberSerialization);

            // only serialize properties that don't start with the specified character
            properties =
                properties.Where(p => !p.PropertyName.StartsWith(_startingWithChar.ToString())).ToList();

            return properties;
        }

        protected override JsonProperty CreateProperty(MemberInfo member, MemberSerialization memberSerialization)
        {
            var property = base.CreateProperty(member, memberSerialization);

            if (property.PropertyName.StartsWith(_startingWithChar))
            {
                property.ShouldSerialize = i => false;
                property.Ignored = true;
            }

            return property;
        }
    }
}

**Comment by elgonzo** [comment]
@shueseman

as i already suggested: _Please also open a new issue report when you are providing your minimal and complete example instead of continuing this thread here_.
